trieste/acquisition/sampler.py

```diff
@@ -276,3 +279,190 @@ def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> TensorTy
         new_order = tf.concat([leading_indices, absolute_trailing_indices], axis=0)
 
         return mean[..., None, :, :] + tf.transpose(variance_contribution, new_order)
+
+
+class ContinuousSampler(ABC):
+    r"""
+    An :class:`ContinuousSampler` samples a specific quantity according
+    to an underlying :class:`ProbabilisticModel`.
+    Unlike our :class:`DiscreteSampler`, :class:`ContinuousSampler` returns a
+    queryable function that provides the sample's value at a query location.
+    """
+
+    def __init__(self, dataset: Dataset, model: ProbabilisticModel):
+        """
+        :param dataset: The data from the observer. Must be populated.
+        :param model: The model to sample from.
+        :raise ValueError: If ``dataset`` is empty.
+        """
+
+        if len(dataset.query_points) == 0:
+            raise ValueError("Dataset must be populated.")
+
+        self._dataset = dataset
+        self._model = model
+
+    def __repr__(self) -> str:
+        """"""
+        return f"{self.__class__.__name__}({self._model!r}, {self._dataset!r})"
+
+    @abstractmethod
+    def sample(self) -> Callable[[TensorType], TensorType]:
```

Comment:
This seems to be an unnecessary level of indirection. This class only has one method, so I would expect that this object is passed around to the place where samples are needed.

I would also expect a method called `sample` to return a sample, not a function for sampling.

Reply:
I don't understand your first point?

Reply:
The flow to get samples is:
`s = Sampler(); sampling_fn = s.sample(); samples = sampling_fn(X)`.
I do not understand why the `sampling_fn` step is needed when it could be:
`s = Sampler(); samples = s.sample(X)`.

Building a smaller object and returning it can make sense when the initial class is very large and has many responsibilities, and it would not make sense to pass the whole thing around. This case is different because the class does nothing else. There is only one method to call on this object.

This would also be a more intuitive behaviour given the method name. Method names on classes can be thought of as the "verbs" that that the object can perform. This "sampler" can "sample", but that produces a "sampling function" not samples.